ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_security.schema.yml

```diff
@@ -24,6 +24,31 @@ keys:
               description: |
                 Local IKE Identification.
                 Can be an IPv4 or an IPv6 address.
+            ike_lifetime:
+              type: int
+              convert_types:
+                - str
+              min: 1
+              max: 24
+              description: |
+                IKE lifetime in hours.
+            encryption:
+              type: str
+              valid_values:
+                - "null"
```

Comment:
Could we call this "no_encryption" to avoid the pitfall with null in yaml?

Reply:
I can, but I did it like this to be consistent with other parts of the schema, which also include "null" , like the sa policies:

```
sa_policies:
        type: list
        description: Security Association policies.
        primary_key: name
        items:
          type: dict
          keys:
            name:
              type: str
              description: Name of the SA policy.
            esp:
              type: dict
              keys:
                integrity:
                  type: str
                  valid_values:
                    - "null"
                    - sha1
                    - sha256
                encryption:
                  type: str
                  valid_values:
                    - "null"
                    - aes128
                    - aes128gcm128
                    - aes128gcm64
                    - aes256
                    - aes256gcm256
```

If I change all 3 to `no_encryption` or `disabled`, it would be a breaking change

Reply:
@gmuloc suggestions? I would like to avoid having a string with `"null"`

Reply:
@gmuloc bump

Reply:
Thanks for the bump @ClausHolbechArista 

so indeed `"null"` may bring some difficulty in proper YAML syntax if users forget quotes - my bad for using it!

@gusmb can you:
1/ add `disabled` valid_values for all of them and adjust the jinja templates to render "null" for disabled
2/ keep `"null"` for the existing ones but move it at the bottom of the list of valid values and add a comment in the description that it is deprecated and we will clean it for 5.0.0 (if you can add a reference  "TODO: AVD 5.0.0" in the schema file in the YAML).

don't hesitate to reach out for help

Reply:
Done

Reply:
I also modified the templates accordingly, since I see that "null" is a valid CLI option for the SA policies, but not for the IKE policies. Therefore, value "disabled" gets translated to "null" in the rendered CLI for those cases